### ISSUE TITLE
ArgumentError in Tzdata update

### ISSUE THREAD

**Issue by ishikawa** [lau/tzdata#46]
Today, We've got this error when startup and sometimes while running:

```
Elixir.ArgumentError: argument error
  Module "ets", in :ets.new/2
  File "lib/tzdata/data_builder.ex", line 11, in Tzdata.DataBuilder.load_and_save_table/0
  File "lib/tzdata/release_updater.ex", line 41, in Tzdata.ReleaseUpdater.poll_for_update/0
  File "lib/tzdata/release_updater.ex", line 17, in Tzdata.ReleaseUpdater.handle_info/2
  File "gen_server.erl", line 616, in :gen_server.try_dispatch/4
  File "gen_server.erl", line 686, in :gen_server.handle_msg/6
  File "proc_lib.erl", line 247, in :proc_lib.init_p_do_apply/3
  Module "Elixir.Tzdata.ReleaseUpdater", in Tzdata.ReleaseUpdater.init/1
```

Arguments passed to `:ets.new/2`

```
arg0	:tzdata_rel_2017c
arg1	[:set, :named_table]
```

We use Tzdata 0.5.12 and Timex 3.1.24.

```
  "timex": {:hex, :timex, "3.1.24", "d198ae9783ac807721cca0c5535384ebdf99da4976be8cefb9665a9262a1e9e3", [:mix], [{:combine, "~> 0.7", [hex: :combine, repo: "hexpm", optional: false]}, {:gettext, "~> 0.10", [hex: :gettext, repo: "hexpm", optional: false]}, {:tzdata, "~> 0.1.8 or ~> 0.5", [hex: :tzdata, repo: "hexpm", optional: false]}], "hexpm"},
  "tzdata": {:hex, :tzdata, "0.5.12", "1c17b68692c6ba5b6ab15db3d64cc8baa0f182043d5ae9d4b6d35d70af76f67b", [:mix], [{:hackney, "~> 1.0", [hex: :hackney, repo: "hexpm", optional: false]}], "hexpm"},
```

**Comment by sickill** [comment]
Started getting the same couple of hours ago.

**Comment by paulswartz** [comment]
Short term fix: `rm -r deps/tzdata/priv/*`

**Comment by ishikawa** [comment]
@paulswartz Certainly we use CircleCI to build our releases and cache `deps` directory. I’ll try to rebuild without cache. Thanks to your advice!

**Comment by lau** [comment]
Thanks everyone for the feedback.

It looks like iana.org is returned a slightly different modified at timestamp for the same 2017c release.
The two different ones I have seen are `Last-Modified: Mon, 23 Oct 2017 15:20:26 GMT` and
`Last-Modified: Mon, 23 Oct 2017 15:19:20 GMT`.

Because of a changed `Last-Modified` timestamp, `Tzdata` expects that there is a new version, and it will download it and try to create a new table for it. But it is the same version, so it crashes when trying to create an already existing ETS table for 2017c.

The crashes are noise, but I expect things to otherwise function as expected. Because it is only the updater that crashes and it should be restarted.

A way to handle this more gracefully and avoid crashing the updater is to check if the newly downloaded version is the same as the already existing version. And if the downloaded version is not a newer version, just stop there, create a log entry about what happened, and not try to create a new ETS table. I hope to have a new version out that handles this more gracefully.

**Comment by lau** [comment]
A new version that fixes the issue has been released: `0.5.13`
